rclcpp_components/include/rclcpp_components/component_manager.hpp

```diff
@@ -256,6 +256,20 @@ class ComponentManager : public rclcpp::Node
     on_list_nodes(request_header, request, response);
   }
 
+  /// Set a manager's executor thread priority level, can throw if priority is not able to be set
+  void setSoftRealTimePriority()
+  {
+    sched_param sch;
+    sch.sched_priority = 49;
+    if (sched_setscheduler(0, SCHED_FIFO, &sch) == -1) {
```

Comment:
I thought that calling this would increase the process's priority and change scheduling strategy, do we need to set all thread's attribute or just the thread which calls this function?

Reply:
The parameter `0` in `sched_setscheduler` means take the current thread. Because `setSoftRealTimePriority` is called in the newly created thread, this function will only set the priority of the newly created thread.

Reply:
> Agreed for the Isolated case. How do we handle this for the normal case where this is simply called while inside of a thread, not in creating one (e.g. we call `setSoftRealTimePriority` in the `ComponentManager` constructor)?
> 
On my thought, the way that to set the current thread's attrbutes takes 2 steps:

1. get the native handle of the current thread
2. set the priority and scheduling strategy of current thread(with native handle)

both needs platform-specific apis.

Reply:
Thanks @JanStaschulat , you are right.
https://man7.org/linux/man-pages/man2/sched_setscheduler.2.html